### PR TITLE
[CI] automate update-beats

### PR DESCRIPTION
**Pull request by v1v**
## Motivation/summary

Jenkins job that:

1) Runs make `update-beats`.
2) Stages the changes to `go.mod`, `go.sum`, `NOTICE.txt`.
3) Creates a PR.

_ONLY_ on daily basis for the `master` branch. If needed, it's possible to trigger it on PR/tag/other-branches using the Jenkins UI.

Mergify does the rest:
- auto-merge 
- delete-on-merge bits set

## How to test these changes

Within the pipeline itself

### Dry Run

image

### No dry run

image

And created https://github.com/elastic/apm-server/pull/5703/files

**NOTE**: Bear in mind the PR was created on the top of this PR, that's the reason for having more changes. When running on the master branch, this should not be the case.

## Related issues

Closes https://github.com/elastic/apm-server/issues/4941
Requires https://github.com/elastic/apm-server/pull/5690

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,551 +1,104 @@
-#!/usr/bin/env groovy
+// Licensed to Elasticsearch B.V. under one or more contributor
```
Comment (v1v) [inline Jenkinsfile:1]: For testing purposes

File: .ci/update-beats.groovy [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  options {
+    timeout(time: 1, unit: 'HOURS')
+    buildDiscarder(logRotator(numToKeepStr: '5', artifactNumToKeepStr: '5'))
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+  }
+  parameters {
+    booleanParam(name: 'DRY_RUN_MODE', defaultValue: true, description: 'If true, allows to execute this pipeline in dry run mode, without sending a PR.')
```
Comment (v1v) [inline .ci/update-beats.groovy:46]: For testing purposes

**Comment by v1v** [comment]
Some errors with the make update-beats

```
16:28:00  diff -u .go-version $(/var/lib/jenkins/workspace/_server_update-beats-mbp_PR-5692/.gvm/versions/go1.16.5.linux.amd64/bin/go list -m -f {{.Dir}} github.com/elastic/beats/v7)/.go-version \
16:28:00  	|| { code=$?; echo ".go-version out of sync with Beats"; exit $code; }
16:28:00  /var/lib/jenkins/workspace/_server_update-beats-mbp_PR-5692/.gvm/versions/go1.16.5.linux.amd64/bin/go build -o build/linux/bin/mage github.com/magefile/mage
16:28:01  ./build/linux/bin/mage -compile=build/linux/mage
16:28:33  # github.com/elastic/apm-server/elasticsearch
16:28:33  elasticsearch/config.go:150:17: assignment mismatch: 2 variables but transport.TLSDialer returns 1 values
16:28:33  Error: error compiling magefiles
16:28:33  Makefile:273: recipe for target 'build/linux/mage' failed
16:28:33  make: *** [build/linux/mage] Error 1
```

File: .ci/update-beats.groovy (hunk 2 above)
Comment (v1v) [reply .ci/update-beats.groovy:46]:
```suggestion
    booleanParam(name: 'DRY_RUN_MODE', defaultValue: false, description: 'If true, allows to execute this pipeline in dry run mode, without sending a PR.')
```

**Review by kuisathaverat — APPROVED** [review]

**Review by axw — APPROVED** [review]
Awesome, thank you very much @v1v! :heart:

**Comment by simitt** [comment]
Amazing, thank you!

**Comment by v1v** [comment]
@Mergifyio backport 7.17

**Pull request merged**